source/views/building-hours/report/overview.tsx

```diff
@@ -23,23 +25,62 @@ import type {
 	NamedBuildingScheduleType,
 	SingleBuildingScheduleType,
 } from '../types'
-import type {TopLevelViewPropsType} from '../../types'
 import {summarizeDays, formatBuildingTimes, blankSchedule} from '../lib'
 import {submitReport} from './submit'
 import {NativeStackNavigationOptions} from '@react-navigation/native-stack'
-import {RouteProp, useNavigation} from '@react-navigation/native'
+import {RouteProp, useNavigation, useRoute} from '@react-navigation/native'
+import {CloseScreenButton} from '@frogpond/navigation-buttons'
 import {RootStackParamList} from '../../../navigation/types'
 
-type Props = TopLevelViewPropsType & {
-	route: {params: {initialBuilding: BuildingType}}
-}
+export let BuildingHoursProblemReportView = (): JSX.Element => {
+	let navigation = useNavigation()
+	let route = useRoute<RouteProp<RootStackParamList, typeof NavigationKey>>()
+	let {initialBuilding} = route.params
+
+	let [building, setBuilding] = React.useState(initialBuilding)
 
-export let BuildingHoursProblemReportView = (props: Props): JSX.Element => {
-	let [building, setBuilding] = React.useState<BuildingType>(
-		props.route.params.initialBuilding,
+	// used for checking against unsaved edits
+	let [hasUnsavedChanges, setHasUnsavedChanges] = React.useState(false)
+	let initialBuildingYaml = React.useMemo(
+		() => jsYaml.dump(initialBuilding),
+		[initialBuilding],
 	)
 
-	let navigation = useNavigation()
+	/**
+	 * checking for unsaved edits
+	 *
+	 * noting that we also have `gestureEnabled` set to false in the navigation options
+	 * (ios only) to prevent dismissing the modal without prompting.
+	 * https://reactnavigation.org/docs/preventing-going-back
+	 */
+	React.useEffect(
+		() =>
+			navigation.addListener('beforeRemove', (event) => {
```

Comment:
Can we save the return value of `addListener`, and removeListener(?) in the cleanup handler of useEffect?